src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetProjectManagerService.cs

```diff
@@ -73,48 +103,384 @@ public async ValueTask<IReadOnlyCollection<PackageReference>> GetInstalledPackag
             return packageReferences.SelectMany(e => e).ToArray();
         }
 
-        public async ValueTask<object> GetMetadataAsync(string projectGuid, string key, CancellationToken cancellationToken)
+        public async ValueTask<object> GetMetadataAsync(string projectId, string key, CancellationToken cancellationToken)
         {
-            var solutionManager = await ServiceLocator.GetInstanceAsync<IVsSolutionManager>();
-            Assumes.NotNull(solutionManager);
+            Assumes.NotNullOrEmpty(projectId);
+            Assumes.NotNullOrEmpty(key);
 
-            NuGetProject project = await GetNuGetProjectMatchingProjectGuidAsync(projectGuid);
+            NuGetProject project = await GetNuGetProjectMatchingProjectIdAsync(projectId);
 
             return project.GetMetadata<object>(key);
         }
 
-        public async ValueTask<(bool, object)> TryGetMetadataAsync(string projectGuid, string key, CancellationToken cancellationToken)
+        public async ValueTask<(bool, object)> TryGetMetadataAsync(string projectId, string key, CancellationToken cancellationToken)
         {
-            NuGetProject project = await GetNuGetProjectMatchingProjectGuidAsync(projectGuid);
+            Assumes.NotNullOrEmpty(projectId);
+            Assumes.NotNullOrEmpty(key);
+
+            NuGetProject project = await GetNuGetProjectMatchingProjectIdAsync(projectId);
 
             bool success = project.TryGetMetadata(key, out object value);
+
             return (success, value);
         }
 
-        public async ValueTask<bool> IsProjectUpgradeableAsync(string projectGuid, CancellationToken cancellationToken)
+        public async ValueTask BeginOperationAsync()
```

Comment:
>BeginOperationAsync [](start = 31, length = 19)

What would happen if this was called by two different projects at the same time?